### ISSUE TITLE
azure_rm_azurefirewall_info fails with KeyError: 'natRuleCollections'\n in simple configuration

### ISSUE THREAD

**Issue by manzomanze** [ansible/ansible_collections_azure#1302]
##### SUMMARY
I wanted to gather info on the firewall network rules, however azure_rm_azurefirewall_info fails with different key errors depending on the parameters involved, i made sure to be able to access the firewall policies resource group and subscriptions.
i'm using az login as the authentication method

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible 2.10.17
  config file = None
  configured module search path = ['/home/kazim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kazim/ansible-azure/lib/python3.11/site-packages/ansible
  executable location = /home/kazim/ansible-azure/bin/ansible
  python version = 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
```

##### COLLECTION VERSION

```paste below
azure.azcollection            1.18.1  
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Running in virtual environment inside an Azure VM
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"

##### STEPS TO REPRODUCE

- hosts: localhost
  connection: local
  tasks:
    - name: Create Azure Firewall App Rule for RedHat resources
      azure.azcollection.azure_rm_azurefirewall_info:
        subscription_id: subscription_id_redacted
        resource_group: rg-ict-network-core-westeu-001
        name: fw-ict-network-core-westeu-001

```- hosts: localhost
  connection: local
  tasks:
    - name: Get facts for one subscription by name
      azure.azcollection.azure_rm_subscription_info:
        name: "subscription_name_redacted"

    - name: Get facts for one resource group
      azure.azcollection.azure_rm_resourcegroup_info:
        name: rg-ict-network-core-westeu-001
        subscription_id: subscription_id_redacted
    - name: Get facts for all firewall policies in resource group
      azure.azcollection.azure_rm_firewallpolicy_info:
        resource_group: rg-ict-network-core-westeu-001
        subscription_id: subscription_id_redacted
        name: fwp-ict-network-core-we-001
    - name: firewall info
      azure.azcollection.azure_rm_azurefirewall_info:
        subscription_id: subscription_id_redacted
        resource_group: rg-ict-network-core-westeu-001
        name: fw-ict-network-core-westeu-001

```
##### EXPECTED RESULTS

Expected  firewall information to be displayed or if i made an error, some kind of error.

##### ACTUAL RESULTS

```
ansible-playbook 2.10.17
  config file = None
  configured module search path = ['/home/kazim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kazim/ansible-azure/lib/python3.11/site-packages/ansible
  executable location = /home/kazim/ansible-azure/bin/ansible-playbook
  python version = 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
No config file found; using defaults
host_list declined parsing /home/kazim/bitbucket/#######/#######/inventory.yml as it did not pass its verify_file() method
script declined parsing /home/kazim/bitbucket/#######/#######/inventory.yml as it did not pass its verify_file() method
Parsed /home/kazim/bitbucket/######/#######/inventory.yml inventory source with yaml plugin
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.

PLAYBOOK: firewall **************************************************************************************************************************************************************************************
1 plays in firewall

PLAY [localhost] ****************************************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************
task path: /home/kazim/bitbucket/#######/#######/firewall:1
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kazim
<localhost> EXEC /bin/sh -c 'echo ~kazim && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/kazim/.ansible/tmp `"&& mkdir "` echo /home/kazim/.ansible/tmp/ansible-tmp-1698393990.6632888-20158-113611149210022 `" && echo ansible-tmp-1698393990.6632888-20158-113611149210022="` echo /home/kazim/.ansible/tmp/ansible-tmp-1698393990.6632888-20158-113611149210022 `" ) && sleep 0'
# removed some logs
ok: [localhost]
META: ran handlers

TASK [Get facts for one subscription by name] ***********************************************************************************************************************************************************
task path: /home/kazim/bitbucket/####/#######/firewall:4
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kazim
# removed some logs
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "all": null,
            "api_profile": "latest",
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "id": null,
            "log_mode": null,
            "log_path": null,
            "name": "Firewall_Subscription_redacted",
            "password": null,
            "profile": null,
            "secret": null,
            "subscription_id": null,
            "tags": null,
            "tenant": null,
            "thumbprint": null,
            "x509_certificate_path": null
        }
    },
    "subscriptions": [
        {
            "display_name": "Firewall_Subscription_redacted",
            "fqid": "/subscriptions/subscription_id_redacted",
            "state": "Enabled",
            "subscription_id": "subscription_id_redacted",
            "tags": null,
            "tenant_id": "tenant_id_redacted"
        }
    ]
}

TASK [Get facts for one resource group] *****************************************************************************************************************************************************************
task path: /home/kazim/bitbucket/#######/#######/firewall:8
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kazim
#ssh logs removed
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "list_resources": null,
            "log_mode": null,
            "log_path": null,
            "name": "rg-ict-network-core-westeu-001",
            "password": null,
            "profile": null,
            "secret": null,
            "subscription_id": "subscription_id_redacted",
            "tags": null,
            "tenant": null,
            "thumbprint": null,
            "x509_certificate_path": null
        }
    },
    "resourcegroups": [
        {
            "id": "/subscriptions/subscription_id_redacted/resourceGroups/rg-ict-network-core-westeu-001",
            "location": "westeurope",
            "name": "rg-ict-network-core-westeu-001",
            "properties": {
                "provisioning_state": "Succeeded"
            },
            "tags": {
                "environment": "core",
                "unit": "network"
            },
            "type": "Microsoft.Resources/resourceGroups"
        }
    ]
}

TASK [Get facts for all firewall policies in resource group] ********************************************************************************************************************************************
task path: /home/kazim/bitbucket/#######/#######/firewall:12
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kazim
# ssh logs removed
ok: [localhost] => {
    "changed": false,
    "firewallpolicies": [
        {
            "base_policy": null,
            "child_policies": [],
            "dns_settings": {
                "enable_proxy": true,
                "require_proxy_for_network_rules": null,
                "servers": [
                    "DC_DNS_redacted_1",
                    "DC_DNS_redacted_2"
                ]
            },
            "etag": "05bea5b4-005a-4345-82ed-e2f48fe4e43a",
            "firewalls": [
                {
                    "id": "/subscriptions/subscription_id_redacted/resourceGroups/rg-ict-network-core-westeu-001/providers/Microsoft.Network/azureFirewalls/fw-ict-network-core-westeu-001"
                }
            ],
            "id": "/subscriptions/subscription_id_redacted/resourcegroups/rg-ict-network-core-westeu-001/providers/Microsoft.Network/firewallPolicies/fwp-ict-network-core-we-001",
            "location": "westeurope",
            "name": "fwp-ict-network-core-we-001",
            "provisioning_state": "Succeeded",
            "rule_collection_groups": [
                {
                    "id": "/subscriptions/subscription_id_redacted/resourcegroups/rg-ict-network-core-westeu-001/providers/Microsoft.Network/firewallPolicies/fwp-ict-network-core-we-001/ruleCollectionGroups/DefaultApplicationRuleCollectionGroup"
                },
                {
                    "id": "/subscriptions/subscription_id_redacted/resourceGroups/rg-ict-network-core-westeu-001/providers/Microsoft.Network/firewallPolicies/fwp-ict-network-core-we-001/ruleCollectionGroups/DefaultNetworkRuleCollectionGroup"
                }
            ],
            "tags": {
                "Environment": "core"
            },
            "threat_intel_mode": "Alert",
            "threat_intel_whitelist": {
                "fqdns": [],
                "ip_addresses": []
            },
            "type": "Microsoft.Network/FirewallPolicies"
        }
    ],
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "log_mode": null,
            "log_path": null,
            "name": "fwp-ict-network-core-we-001",
            "password": null,
            "profile": null,
            "resource_group": "rg-ict-network-core-westeu-001",
            "secret": null,
            "subscription_id": "subscription_id_redacted",
            "tags": null,
            "tenant": null,
            "thumbprint": null,
            "x509_certificate_path": null
        }
    }
}
TASK [firewall info] ************************************************************************************************************************************************************************************
task path: /home/kazim/bitbucket/#######/#######/firewall:16
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kazim

The full traceback is:
Traceback (most recent call last):
  File "/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py", line 102, in <module>
    _ansiballz_main()
  File "/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_azurefirewall_info', init_globals=None, run_name='__main__', alter_sys=True)
  File "<frozen runpy>", line 226, in run_module
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 269, in <module>
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 265, in main
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 141, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 461, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 153, in exec_module
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 190, in get
  File "/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py", line 256, in format_item
KeyError: 'natRuleCollections'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/kazim/.ansible/tmp/ansible-tmp-1698393996.984633-20402-23920059877027/AnsiballZ_azure_rm_azurefirewall_info.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_azurefirewall_info', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 269, in <module>\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 265, in main\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 141, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 461, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 153, in exec_module\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 190, in get\n  File \"/tmp/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload_zcvtp6hc/ansible_azure.azcollection.azure_rm_azurefirewall_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_azurefirewall_info.py\", line 256, in format_item\nKeyError: 'natRuleCollections'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP **********************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

```

**Comment by manzomanze** [comment]
It seems that the problem was due to incorrect tenant and/or subscription id and  the auth_source was set as cli without inverted commas. 
I still wish the error was clearer or at least it suggested what would be the possible problem.

Do i need to close the issue?

**Comment by Fred-sun** [comment]
fixes by #1308
